### DISCUSSION TITLE
datapacks not loading with docker-compose

### DISCUSSION THREAD

**Discussion by Ahptt** [itzg/docker-minecraft-server#1783] [Q&A] [answered]
I've setup mods for this fabric server, and they seem to be loading with no issues. After opening my world file after docker compiles the server, I notice that my mods folder is created under my user ahpt, but the datapacks folder is under root, and has no contents. 

![yaml](https://user-images.githubusercontent.com/107279537/195835279-c432dfa4-5311-4979-ac0d-c5829c3a9d33.png)

I launched docker-compose without detaching it, and the output
`mc-mc-1  | [11:02:54] [main/INFO]: Found new data pack file/Craftable-Bundles-Data-Pack-1.19.2.zip, loading it automatically`
`mc-mc-1  | [11:02:54] [main/INFO]: Found new data pack file/amplified-nether-1-19.zip, loading it automatically`
`mc-mc-1  | [11:02:54] [main/INFO]: Found new data pack file/structory-v1-2.zip, loading it automatically`
`mc-mc-1  | [11:02:54] [main/INFO]: Found new data pack file/terralith-v2-3-5.zip, loading it automatically`
but as soon as the world begins to generate
`mc-mc-1  | [11:03:10] [Worker-Main-15/ERROR]: Couldn't load structure terralith:spire`
and the following crash follows
https://gist.github.com/Ahptt/0db77b77d201e9d7e578025568ba3277

I'm certain I have a config issue with the datapacks, but after reviewing the datapacks section, I can't determine what I'm doing wrong.

![world](https://user-images.githubusercontent.com/107279537/195836484-40d34dcc-588a-44c9-9496-7d3d5dcc7b0c.png)
Any help on this would be greatly appreciated. Thanks!

**Comment by itzg** [comment, marked as answer]
Even though docker doesn't report an error (it should IMHO), you can't overlap mount points.

**Reply by Ahptt** [reply]
How would I then assign the datapack to the correct folder if I can't overlap the mount points? Thank you.

**Reply by itzg** [reply]
Have you looked at

https://github.com/itzg/docker-minecraft-server#datapacks

**Reply by Ahptt** [reply]
Yes, I have. I'm not sure exactly how this is formatted into the docker-compose.yml though. I've been trying to understand it for the past while haha. Does it go in environment?

**Reply by itzg** [reply]
Now that I look closer at the section I do agree it's lacking some detail.

In your compose file
1. Change the third volume mount to `- /home/mc/datapacks:/datapacks` where that choice of container path is totally arbitrary
2. Add `DATAPACKS: /datapacks` to environment

**Reply by Ahptt** [reply]
That worked perfectly! Thank you.
